helper.mac.fish

```diff
@@ -77,22 +81,98 @@ function findRequiredMinMacOS
 end
 
 function opensslVersion
-  set -l oversion $argv[1]
+  set -gx OPENSSL_VERSION $argv[1]
+end
 
-  if test "$oversion" = ""
-    set -e OPENSSL_VERSION
-    return 0
+function downloadOpenSSL
+  if test -z $OPENSSL_VERSION
+    echo "Need OPENSSL_VERSION global variable!"
+    return 1
   end
 
-  switch $oversion
-    case '1.0.2'
-      set -gx OPENSSL_VERSION $oversion
+  set -l directory $WORKDIR/work/openssl
+  set -l url https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz  
+  mkdir -p $directory
+  cd $directory
+  echo "Downloading sources to $directory from URL: $url"
+  curl -LO $url
+  rm -rf openssl-$OPENSSL_VERSION
+  tar -xzvf openssl-$OPENSSL_VERSION.tar.gz
+end
 
-    case '1.1.1'
-      set -gx OPENSSL_VERSION $oversion
+function buildOpenSSL
+  if test "$OPENSSL_SOURCE_DIR" = ""
+    echo "OPENSSL_SOURCE_DIR is not defined. Please download source codes before building OpenSSL!"
+    return 1
+  else if ! test -d $OPENSSL_SOURCE_DIR
+      echo "Directory $OPENSSL_SOURCE_DIR does not exist!"
+      return 1
+  end
+  cd $OPENSSL_SOURCE_DIR
+  mkdir build
+  
+  if test -z "$ARCH"
+    echo "ARCH is not set! Can't decide wether to build OpenSSL for arm64 or x86_64."
+    return 1
+  end
 
+  if test "$ARCH" = "x86_64"
+    set -xg OPENSSL_PLATFORM darwin64-x86_64-cc
+  else if test "$ARCH" = "arm64"
+    set -xg OPENSSL_PLATFORM darwin64-arm64-cc
+  else
+    echo "Unsupported architecture: $ARCH. OpenSSL oskar build is supported for x86_64 or arm64!"
+    return 1
+  end
+
+  for type in shared no-shared
+    for mode in debug release
+      set -l cmd "perl ./Configure --prefix=$OPENSSL_SOURCE_DIR/build/$mode/$type --openssldir=$OPENSSL_SOURCE_DIR/build/$mode/$type/openssl --$mode $type $OPENSSL_PLATFORM"
+      echo "Executing: $cmd"
+      eval $cmd
+      make
+      make test
+      make install_dev
+    end
+  end
+end
+
+function findOpenSSLPath
+  set -gx OPENSSL_SOURCE_DIR $WORKDIR/work/openssl/openssl-$OPENSSL_VERSION
+  set -xg OPENSSL_ROOT $OPENSSL_SOURCE_DIR/build  
+
+  switch $BUILDMODE
+    case "Debug"
+      set mode debug
+    case "Release" "RelWithDebInfo" "MinSizeRel"
+      set mode release
     case '*'
-      echo "unknown openssl version $oversion"
+      echo "Unknown BUILDMODE value: $BUILDMODE. Can't choose OpenSSL build type."
```

Comment:
Please print out available build modes here. And print out how to properly set it

Reply:
Fixed.